pkg/scheduler/objects/application.go

```diff
@@ -695,6 +695,10 @@ func (sa *Application) RecoverAllocationAsk(ask *AllocationAsk) {
 				zap.Error(err))
 		}
 	}
+
+	if ask.IsPlaceholder() {
```

Comment:
Could we extract those behavior to a new method and then let `RecoverAllocationAsk` and `AddAllocationAsk` to reuse that method? That can avoid that `RecoverAllocationAsk` and `AddAllocationAsk` have diverged in the future.

Reply:
Done.